### PR TITLE
sys_fs_fcntl with parameter 0xC0000002 fix

### PR DESCRIPTION
**Pull request by RipleyTom**
sys_fs_fcntl with parameter 0xC0000002 should extract device path from the given path and use that to gather free space information.

I'm open to better implementations :p.

### CODE REVIEW

**Comment by RipleyTom** [comment]
Fixes LBP 1 save issues.

File: rpcs3/Emu/Cell/lv2/sys_fs.cpp [hunk 1]
```diff
@@ -989,13 +989,18 @@ error_code sys_fs_fcntl(u32 fd, u32 op, vm::ptr<void> _arg, u32 _size)
...
 		const std::string_view vpath = arg->path.get_ptr();
-		const std::string local_path = vfs::get(vpath);
 
-		if (vpath.find_first_not_of('/') == -1)
+		if (vpath[0] != '/')
 		{
 			return {CELL_EPERM, vpath};
 		}
 
+		//Extract device from path
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_fs.cpp]: not this again :/

**Comment by Xcedf** [comment]
Fixes same issue, not able to allocate 209 kb for No More Heroes: Paradise

**Comment by digitaldude555** [comment]
Fixes moon diver save.

**Comment by HexyFify** [comment]
can confirm this works

**Comment by Kravickas** [comment]
Fixes 4 access violations in Beyond: Two Souls™ [BCUS98298] while loading save. (Continue)

File: rpcs3/Emu/Cell/lv2/sys_fs.cpp [hunk 2]
```diff
@@ -989,13 +989,21 @@ error_code sys_fs_fcntl(u32 fd, u32 op, vm::ptr<void> _arg, u32 _size)
...
 		{
 			return {CELL_EPERM, vpath};
 		}
 
+		// Extract device from path
+		std::size_t lastslash = vpath.find_first_of('/', 1);
+		if (lastslash == std::string_view::npos) lastslash = vpath.length();
+		else lastslash++;
+
+		const std::string device_path = std::string(vpath).substr(0, lastslash);
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/lv2/sys_fs.cpp]: Now converting string_view to string...

File: rpcs3/Emu/Cell/lv2/sys_fs.cpp [hunk 3]
```diff
@@ -989,13 +989,21 @@ error_code sys_fs_fcntl(u32 fd, u32 op, vm::ptr<void> _arg, u32 _size)
...
+
+		const std::string local_path = vfs::get(device_path);
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/lv2/sys_fs.cpp]: When vfs::get accepts lightweight string_view argument

**Comment by RipleyTom** [comment]
Made the changes.

**Pull request merged**